actors/builtin/market/market_actor.go

```diff
@@ -501,6 +505,22 @@ func (a Actor) CronTick(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
 					AssertMsg(nextEpoch > rt.CurrEpoch() && slashAmount.IsZero(), "deal should not be slashed and should have a schedule for next cron tick"+
 						" as it has not been removed")
 
+					// If this is the first cron tick for the deal, we should  randomize the next epoch
+					// for when it will be processed so an attacker isn't able to schedule too many deals for the same tick.
+					if state.LastUpdatedEpoch == epochUndefined {
```

Comment:
This should be done in PublishStorageDeals instead, thus randomising _all_ processing by cron. As it stands, an attacker can still stack a single cron epoch with a flood of new deals (but then they're distributed after that).

Also please break the logic below out to a free function so it doesn't clutter up whichever method it's called from.

Reply:
This is done.